### PR TITLE
fixed issue #147

### PR DESCRIPTION
**Pull request by pratyakshs**

### CODE REVIEW

File: test/lib_packet_base_test.py [hunk 1]
```diff
@@ -156,7 +157,7 @@ class TestPacketBaseLen(object):
     """
     def test_basic(self):
         packet_base = PacketBase()
-        header = ExtensionHeader(b'data')
+        header = ExtensionHeader(b'da')
```
Comment (pratyakshs) [inline test/lib_packet_base_test.py:160]: the `ExtensionHeader.parse` expects the argument to be exactly 2 bytes. So, changed that.

File: README.md [hunk 2]
```diff
@@ -70,5 +70,5 @@ In order to run the unit tests:
 
 1. `cd test/`
 
-2. `nosetests`
+2. `nosetests3`
```
Comment (tonyo) [inline README.md]: IIUC, `nosetests3` is a part of the `python3-nose` system package, which is not installed by running `./scion.sh deps`. Perhaps we could just give a hint that the Py3 version of `nosetests` should be used here?
The cons of system packages are that they are installed system-wide, updated not that often (in my case `nosetests3` has version 1.3.1, while the latest `nose` from PyPi is 1.3.7) and may vary across distributions.

Comment (pratyakshs) [inline README.md]: @rev112 You're right. Even in the other case though, `nosetests` is a part of `python-nose` package - which is not installed by `./scion.sh deps`. Would you suggest changing this line back to `python3 *.py`?

Comment (pszal) [inline README.md]: I'm confused. Why we cannot just add python3-nose to deps ?

Comment (tonyo) [inline README.md]: `nosetests` is installed together with the `nose` Python package which is listed in `PIP3_DEPS` in `scion.sh` (and hence it's the Py3 version), so there's no need to install `python3-nose`.

So, actually, I don't see any serious reason to change this line in the first place.

Comment (pszal) [inline README.md]: Ok, I see, so @pratyakshs please revert that or elaborate if you don't agree.

Comment (pratyakshs) [inline README.md]: @rev112 I booted from a LiveUSB, and ran `./scion.sh deps`. It doesn't seem to be installing the `nosetests` command. It's of course installing the `nose` module in python3 correctly. IMHO, reverting this line back to `nosetests` isn't correct.

Comment (tonyo) [inline README.md]: @pratyakshs I suspect that you have `nosetests` installed (with the `nose` module), but it's not in your $PATH directories. Check your `~/.local/bin`, if the script is there, just add it to your path.

Comment (pratyakshs) [inline README.md]: @rev112 Agreed, and reverted.

Comment (kormat) [inline README.md]: We _could_ use the python3-nose system package, but we also need 2 nose plugins that aren't packaged by ubuntu, and those would have to be pulled in from pip3 in any case (and i've no idea if they'd work with a system nose package).

I think @shitz is going to add a note to the README about adding `~/.local/bin` to your path.

**Pull request merged**